src/mpid/ch4/src/mpidig_recv.h

```diff
@@ -199,6 +199,15 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_do_irecv(void *buf, MPI_Aint count, MPI_Data
     MPIR_Context_id_t context_id = comm->recvcontext_id + context_offset;
     MPIR_FUNC_ENTER;
 
+    if (*request) {
+        /* ch4-layer pass down a pre-allocated request, let's initialize the mpidig part */
```

Comment:
@hzhou Is there specific change i this patch that fixes the failure for `pt2pt/precv_anysrc`? If so, can you point me to it?

Reply:
In the old code, two branches (line 237 and 255) exit the the function and leave the pre-allocated request uninitialized (the mpidig part). That was the source of failure.

Reply:
Could you document that information somewhere, maybe in the commit message? As this patch combines refactoring and the bug fix, it can be a bit tricky to notice what exactly was fixed. 
(If this would be a simple step for you, a (better) alternative could be to split this patch into two commits - refactor and bugfix.)

Reply:
Sure. Let me try the split

Reply:
done.

Reply:
Thanks. Should the commit message change for this patch now?

Reply:
done.